### PR TITLE
Split README file

### LINKED ISSUE #225: Split readme file
Have something like https://github.com/tock/tock

Create dedicated pages and have links from readme.

Introduction
* Motivation
* A Short Story
* Blog and Tutorial
* Talks
* Crate of the Week

Features
* Key Features
* GUI
* Alternatives

Implementation
* Functionality
* Stack

Documentation

Usage and Development
* Usage
* Build from Source
* Minimum Supported Rust Version

Future and Considerations
* Future
* Performance
* Security
* Considerations

Contribute

### PR DESCRIPTION
**Pull request by IoanaAlexandraO**
fix for https://github.com/radumarias/rencfs/issues/225

### CODE REVIEW

File: docs/Altenatives.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#Alternatives
```
Comment (radumarias) [inline docs/Altenatives.md]: space after `#`

File: docs/Cipher_comparison.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+Cipher comparison
```
Comment (radumarias) [inline docs/Cipher_comparison.md]: add `#`

File: docs/Stack.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+## Stack
```
Comment (radumarias) [inline docs/Stack.md]: just one `#`

**Review by radumarias — CHANGES_REQUESTED** [review]

**Comment by radumarias** [comment]
@IoanaAlexandraO any updates on this?

File: README.md [hunk 4]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
-For the library, you can follow the [documentation](https://docs.rs/rencfs/latest/rencfs/).
 
-## Command Line Tool
 
-### Dependencies
+# Introduction
 
-To use the encrypted file system, you need to have FUSE installed on your system. You can install it by running the
-following command (or based on your distribution).
+- Motivation
```
Comment (radumarias) [inline README.md:37]: use `##`

File: README.md [hunk 5]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-Arch
+  Create a `simple,` `performant,` `modular` and `ergonomic` yet `very secure` `encrypted filesystem` to protect your `privacy`, which is also `open source` and is correctly and safely using `well-known audited` crates as `cryptographic primitives.`
 
-```bash
-sudo pacman -Syu && sudo pacman -S fuse3
-```
+- A short story
```
Comment (radumarias) [inline README.md:41]: use `##`

Comment (radumarias) [reply README.md:41]: and for all `-` in main `#` use sub-chapter levels like `##`, `###`...

File: README.md [hunk 6]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
-logged out, or it may not succeed at all.
 
-If, after installation, running `rustc --version` in the console fails, this is the most likely reason.
-In that case please add it to the `PATH` manually.
 
-The project is set up to use the `nightly` toolchain in `rust-toolchain. tool`; on the first build, you will see it fetch the nightly.
+# Features 
 
-Make sure to add this to your `$PATH` too
+- [KEY features](docs/Key_features.md)
```
Comment (radumarias) [inline README.md]: `# Key...`

File: README.md [hunk 7]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-```bash
-export PATH="$PATH::$HOME/.cargo/bin"
-```
+- GUI
```
Comment (radumarias) [inline README.md]: remove

File: README.md [hunk 8]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-```bash
-cargo install cargo-aur
-cargo install cargo-generate-rpm
-```
+There is a [GUI](https://github.com/radumarias/rencfs-desktop/blob/main/demo.gif) too.
 
-### Other dependencies
+- [Alternatives](docs/Alternatives.md)
```
Comment (radumarias) [inline README.md:84]: use `#`

File: README.md [hunk 9]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
-sudo dnf update && sudo dnf install fuse3 && dnf install @development-tools act
-```
 
-### Build for debug
+# Documentation
 
-```bash
-cargo build
-```
+- Docs
```
Comment (radumarias) [inline README.md]: link to `docs` folder

File: README.md [hunk 10]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
-cargo build --release
-```
+For detailed description of the various sequence flows please look into [Flows](docs/flows.md).
 
-### Run
 
-```bash
-cargo run --release -- mount --mount-point MOUNT_POINT --data-dir DATA_DIR
-```
+- What separates us
```
Comment (radumarias) [inline README.md]: move to `Alternatives` from above as `## What separates us`

File: docs/Key_features.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+## Key features
+
+Some of these are still being worked on and marked with `[WIP]`.
+
+- `Security` using well-known audited `AEAD` cryptography primitives;
```
Comment (radumarias) [inline docs/Key_features.md:4]: Replace the lines from here in the bigger list. Don't delete the list just replace corresponding ones

```
- [WIP] [Data integrity, data is written with WAL to ensure integrity even on crash or power loss](https://github.com/radumarias/rencfs/issues/48)
- [WIP] [Hide all info for enhanced privacy; all metadata, content, file name, file size, *time fields, files count, and directory structure is encrypted](https://github.com/radumarias/rencfs/issues/53)
- Safely manage credentials in memory with mlock(2), mprotect, zeroize, and expiry to mitigate cold boot attacks;
- [WIP] [Generate unique nonce in offline mode](https://github.com/radumarias/rencfs/issues/47)
- [WIP] [Add file inode and chunk index to AAD](https://github.com/radumarias/rencfs/issues/49) This prevents blocks from being copied between or within files by an attacker.
- Fast seek on both reads and writes;
- Exposed with FUSE;
- [WIP] [Handle long file names](https://github.com/radumarias/rencfs/issues/47)
- [WIP] [Abstraction layer for Rust File and fs API to use it as lib to switch to using encrypted files by just changing the use statements](https://github.com/radumarias/rencfs/issues/97)
- [WIP] [Abstraction layer to access the storage with implementations for desktop, Wasm, Android, and iOS and the ability to write your own implementation](https://github.com/radumarias/rencfs/issues/111)
```

File: docs/Altenatives.md (hunk 1 above)
Comment (radumarias) [inline docs/Altenatives.md]: rename the file and add missing `r` `Alternatives.md`

**Review by radumarias — CHANGES_REQUESTED** [review]

File: docs/Key_features.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
+## Key features
```
Comment (radumarias) [inline docs/Key_features.md:1]: move these directly in readme file as these are selling points, are important for users to see

**Review by radumarias — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-#### Install and run local RPM
 
-```bash
-cd target/generate-rpm/
-sudo dnf localinstall rencfs-xxx.x86_64.rpm
-```
 
-## Developing inside a Container
+# Future and Considerations
```
Comment (radumarias) [inline README.md]: `# Next steps`

File: README.md [hunk 14]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-See here how to configure for [RustRover](https://www.jetbrains.com/help/rust/connect-to-devcontainer.html) and for [VsCode](https://code.visualstudio.com/docs/devcontainers/containers).
+- Future
```
Comment (radumarias) [inline README.md]: remove

File: README.md [hunk 15]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-You can use the `.devcontainer` directory from the project to start a container with all the necessary tools to build
-and run the app.
+    The plan is to implement it also on macOS and Windows
```
Comment (radumarias) [inline README.md]: move in list

File: README.md [hunk 16]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
+   - **Systemd service** is being worked on [rencfs-daemon](https://github.com/radumarias/rencfs-daemon)
+   - **GUI** is being worked on [rencfs-desktop](https://github.com/radumarias/rencfs-desktop) and [rencfs-kotlin](https://github.com/radumarias/rencfs-kotlin)
+   - **Mobile apps** for **Android** and **iOS** are being worked on [rencfs-kotlin](https://github.com/radumarias/rencfs-kotlin)
 
-# Minimum Supported Rust Version (MSRV)
+- Performance
```
Comment (radumarias) [inline README.md]: Add `# Considerations`
and `## Performance`

File: README.md [hunk 17]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
+on most CPUs via AES-NI. However, where hardware acceleration is not available, `ChaCha20Poly1305` is faster. Also `ChaChaPoly1305` is better at `SIMD`.
 
-# Future
+- [Cipher comparison](docs/Cipher_comparison.md)
 
-The plan is to implement it also on macOS and Windows
-- **Systemd service** is being worked on [rencfs-daemon](https://github.com/radumarias/rencfs-daemon)
-- **GUI** is being worked on [rencfs-desktop](https://github.com/radumarias/rencfs-desktop) and [rencfs-kotlin](https://github.com/radumarias/rencfs-kotlin)
-- **Mobile apps** for **Android** and **iOS** are being worked on [rencfs-kotlin](https://github.com/radumarias/rencfs-kotlin)
+- [⚠️ Security ](docs/Security.md)
```
Comment (radumarias) [inline README.md:136]: move after `Performance`

File: README.md [hunk 18]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
 
-# Performance
+- [Considerations](docs/Considerations.md)
```
Comment (radumarias) [inline README.md]: rename to `Others`

File: README.md [hunk 19]
```diff
@@ -24,514 +24,112 @@ You can also store it in a cloud storage service like Google Drive, Dropbox, etc
...
-```bash
-cd mnt
-mkdir a && cd a
-echo "test" > test.txt
-cat test.txt
-```
+[The Hitchhiker’s Guide to Building an Encrypted Filesystem in Rust](docs/The_Hitchhiker_s_Guide_to_Building_an_Encrypted_Filesystem_in_Rust_2.pdf)
 
-## Locally
+- Blog and tutorial
```
Comment (radumarias) [inline README.md]: Move to the bottom like
```
# Follow us

## Blog and tutorial
```

**Review by radumarias — CHANGES_REQUESTED** [review]

File: docs/Alternatives.md [hunk 20]
```diff
@@ -15,3 +15,8 @@
 - [File Lock PEA](https://eck.cologne/peafactory/en/html/file_pea.html)
```
Comment (radumarias) [inline docs/Alternatives.md:15]: where are the rest? See here the full list https://github.com/radumarias/rencfs?tab=readme-ov-file#alternatives

**Review by radumarias — CHANGES_REQUESTED** [review]

File: docs/Alternatives.md (hunk 20 above)
Comment (IoanaAlexandraO) [reply docs/Alternatives.md:15]: There are 15 bullets both sides.

**Review by radumarias — APPROVED** [review]

**Pull request merged**